src/workers/notifications/worker.ts

```diff
@@ -11,12 +12,48 @@ export type NotificationHandlerReturn =
     }[]
   | undefined;
 
+export type NotificationWorkerHandler = (
+  message: Message,
+  con: DataSource,
+  logger: FastifyBaseLogger,
+) => Promise<NotificationHandlerReturn>;
+
 export interface NotificationWorker {
   subscription: string;
+  handler: NotificationWorkerHandler;
+  maxMessages?: number;
+}
+
+interface TypedNotificationWorkerProps<
```

Comment:
Would like to have `TypedWorker` used here, we can extend with return type something like `interface TypedWorker<T extends keyof PubSubSchema, R = void>
  extends BaseTypedWorker<PubSubSchema[T], R> {}`

but I am ok with this as interim soluion.

Reply:
Maybe just note it down somewhere in web board so we don't forget

Reply:
Yeah, I think it would be better to ultimately originate from the same type which is the `TypedWorker`. Just did this for a quick setup we can follow through. Let me write it down on the board.